### PR TITLE
Add tests checking NM plugins installed

### PR DESCRIPTION
**Pull request by liangwen12year**
Signed-off-by: Wen Liang <liangwen12year@gmail.com>

### CODE REVIEW

File: tests/playbooks/tests_team_exists.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          address:
+            - "192.0.2.24/24"
+            - "2001:db8::23/64"
+
+  tasks:
+      - name: Get NetworkManager-team RPM version
+        command:
+          cmd: rpm -qa --qf '%{name}-%{version}-%{release}\n' NetworkManager-team
+          warn: false
+        register: __rpm_q_NetworkManager
```
Comment (ffmancera) [inline tests/playbooks/tests_team_exists.yml:22]: This is gathering name-version-release but not checking that the package exists.

**Review by ffmancera — CHANGES_REQUESTED** [review]
Hello! Thank you for this contribution. This is missing tests for the wifi plugin and in addition `ensure_provider_tests.py` needs to be modified too.

File: tests/playbooks/tests_team_exists.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      when: "'NetworkManager-team' not in ansible_facts.packages"
+
+    - name: "Team interface configuration"
+      include_role:
+        name: linux-system-roles.network
+      vars:
+        network_connections:
+          # Specify the team profile
+          - name: team0
+            state: up
```
Comment (tyll) [inline tests/playbooks/tests_team_exists.yml:26]: Does this need state: up? Using only persistent_state:present might allow to test this also on Fedora.

File: tests/playbooks/tests_team_exists.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            type: team
+            interface_name: team0
+            # ip configuration (optional)
```
Comment (tyll) [inline tests/playbooks/tests_team_exists.yml:29]: If it is optional, why do you specify it?

**Comment by tyll** [comment]
Thanks, please check the CI failures. Please verify that the test fails when the role does not install the expected packages.

File: tests/playbooks/tests_wireless_exists.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+              password: "p@55w0rD"
+
+    - name: "get the rpm package facts"
+      package_facts:
+        manager: "auto"
+
+    - name: "NetworkManager-wifi checking after wireless configuration"
+      debug:
+        msg: "NetworkManager-wifi is installed after wireless configuration"
+      when: "'NetworkManager-wifi' in ansible_facts.packages"
```
Comment (ffmancera) [inline tests/playbooks/tests_wireless_exists.yml:40]: IMO, this should be an assert not a debug message.

File: tests/playbooks/tests_team_exists.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                - "2001:db8::23/64"
+
+    - name: "get the rpm package facts"
+      package_facts:
+        manager: "auto"
+
+    - name: "NetworkManager-team checking after team configuration"
+      debug:
+        msg: "NetworkManager-team is installed after team configuration"
+      when: "'NetworkManager-team' in ansible_facts.packages"
```
Comment (ffmancera) [inline tests/playbooks/tests_team_exists.yml:42]: This should be an assert not a debug message. If the package is not installed it should fail which is not happening now.

**Review by ffmancera — CHANGES_REQUESTED** [review]

File: tests/playbooks/tests_wireless_exists.yml (hunk 4 above)
Comment (liangwen12year) [reply tests/playbooks/tests_wireless_exists.yml:40]: thanks for hint

File: tests/playbooks/tests_team_exists.yml (hunk 5 above)
Comment (liangwen12year) [reply tests/playbooks/tests_team_exists.yml]: updated

File: tests/playbooks/tests_team_exists.yml (hunk 3 above)
Comment (liangwen12year) [reply tests/playbooks/tests_team_exists.yml:30]: removed, originally I was thinking to provide a sample

File: tests/ensure_provider_tests.py [hunk 6]
```diff
@@ -72,11 +72,17 @@
     "playbooks/tests_team.yml": {
         EXTRA_RUN_CONDITION: "ansible_distribution != 'Fedora'",
     },
+    "playbooks/tests_team_exists.yml": {
+        EXTRA_RUN_CONDITION: "ansible_distribution != 'Fedora'",
```
Comment (tyll) [inline tests/ensure_provider_tests.py:76]: Why is this needed?

File: tests/ensure_provider_tests.py [hunk 7]
```diff
@@ -72,11 +72,17 @@
...
+    },
     # mac80211_hwsim (used for tests_wireless) only seems to be available
     # and working on RHEL/CentOS 7
     "playbooks/tests_wireless.yml": {
         EXTRA_RUN_CONDITION: "ansible_distribution_major_version == '7'",
     },
+    "playbooks/tests_wireless_exists.yml": {
+        EXTRA_RUN_CONDITION: "ansible_distribution_major_version == '7'",
```
Comment (tyll) [inline tests/ensure_provider_tests.py:84]: Why is this needed?

File: tests/playbooks/tests_team_exists.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: BSD-3-Clause
```
Comment (tyll) [inline tests/playbooks/tests_team_exists.yml:1]: Please adjust the name to something like "tests_team_plugin_installation.yml" and the wireless test as well.

File: tests/ensure_provider_tests.py [hunk 9]
```diff
@@ -72,11 +72,17 @@
     "playbooks/tests_team.yml": {
```
Comment (tyll) [inline tests/ensure_provider_tests.py:72]: The commit message does not seem to be correct English. How about "Test installation of NetworkManager plugins"?

File: tests/playbooks/tests_team_exists.yml (hunk 2 above)
Comment (liangwen12year) [reply tests/playbooks/tests_team_exists.yml]: I updated to using `persistent_state:present`

File: tests/playbooks/tests_team_exists.yml (hunk 8 above)
Comment (liangwen12year) [reply tests/playbooks/tests_team_exists.yml]: updated, thanks

File: tests/ensure_provider_tests.py (hunk 9 above)
Comment (liangwen12year) [reply tests/ensure_provider_tests.py:72]: updated, thanks

File: tests/ensure_provider_tests.py (hunk 6 above)
Comment (liangwen12year) [reply tests/ensure_provider_tests.py]: tested, it is not needed, updated the commit already

File: tests/ensure_provider_tests.py (hunk 7 above)
Comment (liangwen12year) [reply tests/ensure_provider_tests.py]: updated, thanks

File: tests/playbooks/tests_team_exists.yml (hunk 8 above)
Comment (liangwen12year) [reply tests/playbooks/tests_team_exists.yml]: Hi Till, the CI failed because the following two line are too long(> 80 char):

./tests/tests_wireless_plugin_installation_nm.yml
  6:81      error    line too long (91 > 80 characters)  (line-length)

./tests/tests_team_plugin_installation_nm.yml
  6:81      error    line too long (87 > 80 characters)  (line-length)
```
  name: Run playbook 'playbooks/tests_wireless_plugin_installation.yml' with nm as provider
  name: Run playbook 'playbooks/tests_team_plugin_installation.yml' with nm as provider
```

Comment (liangwen12year) [reply tests/playbooks/tests_team_exists.yml]: @tyll , shall we change the file name to be the previous name, like "tests_team_exists.yml" ?

Comment (liangwen12year) [reply tests/playbooks/tests_team_exists.yml]: diable the CI check for these two files

Comment (tyll) [reply tests/playbooks/tests_team_exists.yml]: It would be best to disable the line length checks for the generated files or at least those lines. Another option would be to break the lines but that might be too complicated.

Comment (liangwen12year) [reply tests/playbooks/tests_team_exists.yml]: @tyll , updated the commit

**Comment by liangwen12year** [comment]
@ffmancera , do you know how can see the conflicts?

File: tests/playbooks/tests_wireless_plugin_installation.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        name: linux-system-roles.network
+      vars:
+        network_connections:
+          - name: wlan0
+            type: wireless
+            wireless:
+              ssid: "My WPA2-PSK Network"
+              key_mgmt: "wpa-psk"
+            # recommend vault encrypting the wireless password
+            # see https://docs.ansible.com/ansible/latest/user_guide/vault.html
```
Comment (ffmancera) [inline tests/playbooks/tests_wireless_plugin_installation.yml:31]: Just a nit, this is not indented correctly.

**Review by ffmancera — CHANGES_REQUESTED** [review]

File: tests/playbooks/tests_wireless_plugin_installation.yml (hunk 10 above)
Comment (liangwen12year) [reply tests/playbooks/tests_wireless_plugin_installation.yml]: removed

**Review by cathay4t — APPROVED** [review]

**Comment by cathay4t** [comment]
Force merging as the CI is failed because http error when installing epel7 rpm.
